presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -97,12 +99,12 @@ public CacheStatsMBean getCursorCacheStats()
         };
     }
 
-    public Supplier<PageProcessor> compilePageProcessor(Optional<RowExpression> filter, List<? extends RowExpression> projections)
+    public Supplier<PageProcessor> compilePageProcessor(Optional<RowExpression> filter, List<? extends RowExpression> projections, Optional<PlanNodeId> planNodeId, Optional<StageId> stageId)
```

Comment:
Instead of having the page compiler take specific types of `PlanNodeId` and `StageId`, I would prefer a simpler `Optional<String> classNamePrefix` to make it clear to the caller that we don't *need* these specific values.